src/lsm/superblock_free_set.zig

```diff
@@ -175,12 +188,32 @@ pub const SuperBlockFreeSet = struct {
     }
 
     /// Returns the number of bytes written to `target`.
+    /// The encoded data does *not* incldue staged changes.
     pub fn encode(set: SuperBlockFreeSet, target: []align(@alignOf(usize)) u8) usize {
         assert(target.len == SuperBlockFreeSet.encode_size_max(set.blocks.bit_length));
 
         return ewah.encode(bitset_masks(set.blocks), target);
     }
 
+    /// Returns the number of bytes written to `target`.
+    /// The encoded data *does* include staged changes.
+    pub fn encode_with_staging(set: *SuperBlockFreeSet, target: []align(@alignOf(usize)) u8) usize {
+        const count_free = set.count_released();
+        const count_staged = set.staging.count();
+        assert(target.len == SuperBlockFreeSet.encode_size_max(set.blocks.bit_length));
+
+        // Temporarily mark the staged blocks as free.
+        set.blocks.toggleSet(set.staging);
+        assert(set.blocks.count() == count_free + count_staged);
```

Comment:
Nice assertion to have.

Reply:
Since staging is so powerful here, with the ability to toggle all bits if staging is accidentally initialized the wrong way, it would be good to assert in `init()` that `staging.count() == 0`.